core/src/main/java/com/yahoo/tagchowder/Parser.java

```diff
@@ -1185,6 +1195,14 @@ private String makeName(final char[] buff, final int offset, final int length) {
                 }
                 start = true;
                 dst.append(translateColons ? '_' : ch);
+            } else if (ampValidation) {
+                if (start && (ch == '[' || ch == '{' || ch == '⚡'))  {
```

Comment:
based on your code, you are going to accept and not sanitize following html? These examples are not full cases. Please test it fully and have document expected behavior and add corresponding UTs.

<p [text}="cart.p1_qty" class="qty_num">1</p>
<p {text}="cart.p1_qty" class="qty_num">1</p>
<p {text]="cart.p1_qty" class="qty_num">1</p>
<p {text="cart.p1_qty" class="qty_num">1</p>
<p text}="cart.p1_qty" class="qty_num">1</p>
<p [text="cart.p1_qty" class="qty_num">1</p>
<p text]="cart.p1_qty" class="qty_num">1</p>
<p text]="cart.p1_qty" class="qty_num">1</p>
<p ⚡text]="cart.p1_qty" class="qty_num">1</p>
<p ⚡text}="cart.p1_qty" class="qty_num">1</p>
<p ⚡text="cart.p1_qty" class="qty_num">1</p>
<p ⚡text="cart.p1_qty" class="qty_num">1</p>

Reply:
Redo the implementation to ensure brackets are balanced and check for "⚡4email" validity.